exist-core/src/main/java/org/exist/http/servlets/AbstractExistHttpServlet.java

```diff
@@ -246,9 +246,9 @@ protected Subject authenticate(HttpServletRequest request, HttpServletResponse r
 	        }
         }
 
-        // Secondly try basic authentication
+        // Secondly try basic authentication if there is no Authorization header or the Authorization header does not indicate Basic auth
         final String auth = request.getHeader("Authorization");
-        if (auth == null && getDefaultUser() != null) {
```

Comment:
Would it make sense to extract the `getDefaultUser()` value to a local variable?

Reply:
I think as there are only 2 calls and the underlying function implementation just accesses a known class member it will make little difference... also I guess the compiler will do some optimisation anyway?